apps/prairielearn/src/pages/instructorInstanceAdminLti/instructorInstanceAdminLti.html.ts

```diff
@@ -0,0 +1,304 @@
+                      </strong>
+                    </p>
+                    ${!config.hasLti ? html`<p><em>LTI not enabled on this server.</em></p>` : ''}
+                  </div>
+                </div>
+
+                ${config.hasLti
+                  ? html`
+                      ${LtiCredentialsCard({ lti_credentials, csrfToken })}
+                      ${LtiLinkTargetsCard({ lti_links, assessments, csrfToken })}
+                    `
+                  : ''}
+              `}
+        </main>
+      </body>
+    </html>
+  `.toString();
+}
+
+function LtiCredentialsCard({
+  lti_credentials,
+  csrfToken,
+}: {
+  lti_credentials: (Omit<LtiCredentials, 'deleted_at'> & {
+    created: string | null;
+    deleted: string | null;
+  })[];
+  csrfToken: string;
+}) {
+  return html`
+    <div class="card mb-4">
+      <div class="card-header bg-primary text-white">LTI credentials</div>
+      <div class="card-body">
+        <p class="mb-0">
+          Use these credentials with your LMS (Learning Management System) to connect into this
+          course instance. A single credential can be shared with multiple links.
+        </p>
+      </div>
+
+      <table class="table table-sm table-hover">
+        <thead>
+          <tr>
+            <th>Launch URL</th>
+            <th>Consumer key</th>
+            <th>Shared secret</th>
+            <th>Created</th>
+            <th>Deleted</th>
+          </tr>
+        </thead>
+        <tbody>
+          ${lti_credentials
+            ? lti_credentials.map(
+                (tc) => html`
+                  <tr>
+                    <td>
+                      <code>${config.ltiRedirectUrl}</code>
+                      <i
+                        class="far fa-copy"
+                        title="Copy to clipboard"
+                        onClick="copyToClipboard($(this).prev());$(this).fadeOut({queue: true});$(this).fadeIn({queue:true});"
+                      ></i>
+                    </td>
+                    <td>
+                      <code>${tc.consumer_key}</code>
+                      <i
+                        class="far fa-copy"
+                        title="Copy to clipboard"
+                        onClick="copyToClipboard($(this).prev());$(this).fadeOut({queue: true});$(this).fadeIn({queue:true});"
+                      ></i>
+                    </td>
+                    <td>
+                      <code>${tc.secret}</code>
+                      <i
+                        class="far fa-copy"
+                        title="Copy to clipboard"
+                        onClick="copyToClipboard($(this).prev());$(this).fadeOut({queue: true});$(this).fadeIn({queue:true});"
+                      ></i>
+                    </td>
+                    <td>${tc.created}</td>
+                    <td>
+                      ${tc.deleted ||
+                      html`
+                        <a
```

Comment:
I'd say "just use a `<button>`" if there wasn't the caveat about you skipping some common modernization things.

Reply:
Yeah, in this case I'll rely on "Just Works :tm:" and merge this without changing it so we can move on with partials.